src/Illuminate/Support/Number.php

```diff
@@ -0,0 +1,161 @@
+<?php
+
+namespace Illuminate\Support;
+
+use Illuminate\Support\Traits\Macroable;
+use NumberFormatter;
+use RuntimeException;
+
+class Number
+{
+    use Macroable;
+
+    /**
+     * The current default locale.
+     *
+     * @var string
+     */
+    protected static $locale = 'en';
+
+    /**
+     * Format the given number according to the current locale.
+     *
+     * @param  int|float  $number
+     * @param  ?string  $locale
+     * @return string|false
+     */
+    public static function format(int|float $number, ?string $locale = null)
+    {
+        static::ensureIntlExtensionIsInstalled();
+
+        $formatter = new NumberFormatter($locale ?? static::$locale, NumberFormatter::DECIMAL);
+
+        return $formatter->format($number);
+    }
+
+    /**
+     * Convert the given number to its percentage equivalent.
+     *
+     * @param  int|float  $number
+     * @param  int  $precision
+     * @param  ?string  $locale
+     * @return string|false
+     */
+    public static function toPercentage(int|float $number, int $precision = 0, ?string $locale = null)
+    {
+        static::ensureIntlExtensionIsInstalled();
+
+        $formatter = new NumberFormatter($locale ?? static::$locale, NumberFormatter::PERCENT);
+
+        $formatter->setAttribute(NumberFormatter::FRACTION_DIGITS, $precision);
+
+        return $formatter->format($number / 100);
+    }
+
+    /**
+     * Convert the given number to its currency equivalent.
+     *
+     * @param  int|float  $number
+     * @param  string  $currency
+     * @param  ?string  $locale
+     * @return string|false
+     */
+    public static function toCurrency(int|float $number, string $currency = 'USD', ?string $locale = null)
+    {
+        static::ensureIntlExtensionIsInstalled();
+
+        $formatter = new NumberFormatter($locale ?? static::$locale, NumberFormatter::CURRENCY);
+
+        return $formatter->formatCurrency($number, $currency);
+    }
+
+    /**
+     * Convert the given number to its file size equivalent.
+     *
+     * @param  int|float  $bytes
+     * @param  int  $precision
+     * @return string
+     */
+    public static function toFileSize(int|float $bytes, int $precision = 0)
+    {
+        $units = ['B', 'KB', 'MB', 'GB', 'TB', 'PB', 'EB', 'ZB', 'YB'];
+
+        for ($i = 0; ($bytes / 1024) > 0.9 && ($i < count($units) - 1); $i++) {
+            $bytes /= 1024;
+        }
+
+        return sprintf('%s %s', number_format($bytes, $precision), $units[$i]);
+    }
+
+    /**
+     * Convert the number to its human readable equivalent.
+     *
+     * @param  int  $number
+     * @param  int  $precision
+     * @return string
+     */
+    public static function forHumans(int|float $number, int $precision = 0)
```

Comment:
What about other languages?

Reply:
Hmm, maybe the `__()` function could be used to provide translation support?

Reply:
https://github.com/laravel/framework/pull/48845#issuecomment-1817535149